boot/zephyr/Kconfig

```diff
@@ -346,4 +346,15 @@ config UPDATEABLE_IMAGE_NUMBER
 	help
 	  Enables support of multi image update.
 
+config MCUBOOT_DOWNGRADE_PREVENTION
+	bool "Downgrade prevention"
+	depends on BOOT_UPGRADE_ONLY
+	help
+	  Prevent downgrades by enforcing incrementing version numbers.
+	  When this option is set, any upgrade must have greater major version
+	  or greater minor version with equal major version. For this mechanism
+	  to be trustworthy, the application must enable flash protection of
+	  the primary partition. Otherwise, the downgrade prevention can be
+	  circumvented by erasing the header of the primary slot.
```

Comment:
I am OK with merging this PR, but this comment does not seem correct regarding security implications. Do you plan to fix it @hakonfam?

Reply:
btw, https://github.com/JuulLabs-OSS/mcuboot/pull/628#issuecomment-577825787 is a good suggestion!